### DISCUSSION TITLE
Simulating dispersive metal in the THz regime

### DISCUSSION THREAD

**Discussion by dheligman** [NanoComp/meep#1922] [Q&A]
Hi All,

I have been using MEEP for my FDTD needs for several years now. It wasn't until now that I need to simulate a dispersive metal in the range between 0.1 - 2 THz. I am having trouble figuring out how to produce a stable simulation for this. I am specifically trying to simulate gold if that helps.

Best,
Daniel

**Comment by oskooi** [comment]
Have you tried any of the six remedies for field instabilities described in https://meep.readthedocs.io/en/latest/FAQ/#why-are-the-fields-blowing-up-in-my-simulation? In particular, for dispersive materials such gold, you may need to replace the `PML` with an `Absorber` and/or increase the grid resolution.

**Reply by dheligman** [reply]
Hi oskooi,

I looked through this one person's work, specifically their meep_materials library. His GitHub page is https://github.com/FilipDominec/python-meep-utils. He was talking about how in the THz regime you get a very large negative peittivity for metals especially gold. He mentioned something about a critical frequency and that the permittivity must be positive above it in order for the simulation to be stable. And I have tried higher resolution, smaller Courant, and switching my PMLs to absorbers but I can't seem to eliminate the instability. 

Best,
Daniel

**Comment by dalarev** [comment]
Are you perhaps using periodic boundary conditions and symmetries? I believe there is a [known issue ](https://github.com/NanoComp/meep/discussions/1911#discussioncomment-1999005)with the combination.
